File: spring/fizz-buzz/src/test/java/de/welcz/fizzbuzz/FizzBuzzServiceTest.java
package de.welcz.fizzbuzz;

import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;
import org.junit.jupiter.params.provider.ValueSource;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@ExtendWith(SpringExtension.class)
class FizzBuzzServiceTest {
    @Autowired
    private FizzBuzzService service;

    @ParameterizedTest
    @CsvSource(value = {
            "1, 1",
            "2, 2",
    })
    void replicates_normal_numbers(int input, String expected) {
        assertThat(service.calculate(input)).isEqualTo(expected);
    }

    @ParameterizedTest
    @ValueSource(ints = {3, 6})
    void fizz_numbers_divisible_by_3(int input) {
        assertThat(service.calculate(input)).isEqualTo("Fizz");
    }

    @ParameterizedTest
    @ValueSource(ints = {5, 10})
    void buzz_numbers_divisible_by_5(int input) {
        assertThat(service.calculate(input)).isEqualTo("Buzz");
    }

    @ParameterizedTest
    @ValueSource(ints = {15, 30})
    void fizz_buzz_numbers_divisible_by_3_and_5(int input) {
        assertThat(service.calculate(input)).isEqualTo("Fizz-Buzz");
    }
}

File: kafka/fizzbuzz-kafka-streams/src/test/kotlin/FizzBuzzServiceTest.kt
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class FizzBuzzServiceTest {
    private lateinit var fizzBuzzService: FizzBuzzService

    @BeforeEach
    internal fun setUp() {
        fizzBuzzService = FizzBuzzService()
    }

    @ParameterizedTest
    @ValueSource(ints = [1, 2, 4])
    internal fun `replicate default number`(input: Int) {
        assertThat(fizzBuzzService.single(input)).isEqualTo(input.toString())
    }

    @ParameterizedTest
    @ValueSource(ints = [3, 6, 9])
    internal fun `replace number divisible by 3 with 'Fizz'`(input: Int) {
        assertThat(fizzBuzzService.single(input)).isEqualTo("Fizz")
    }

    @ParameterizedTest
    @ValueSource(ints = [5, 10])
    internal fun `replace number divisible by 5 with 'Buzz'`(input: Int) {
        assertThat(fizzBuzzService.single(input)).isEqualTo("Buzz")
    }

    @ParameterizedTest
    @ValueSource(ints = [15, 30])
    internal fun `replace number divisible by 3 and 5 with 'Fizz-Buzz'`(input: Int) {
        assertThat(fizzBuzzService.single(input)).isEqualTo("Fizz-Buzz")
    }

    @Test
    internal fun `create expected sequence`() {
        assertThat(fizzBuzzService.sequenceUpTo(15)).containsExactly(
                "1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8",
                "Fizz", "Buzz", "11", "Fizz", "13", "14", "Fizz-Buzz")
    }
}

